### PR TITLE
fix: [CI-6485]: Support disabledExpiry Watcher for qa

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42151)

### CODE REVIEW

**Comment by smjt-h** [comment]
trigger messagecheck

File: product/log-service/stream/redis/redis.go [hunk 1]
```diff
@@ -44,8 +45,9 @@ const (
 	entryKey       = "line"
 
 	// Redis TTL error values
-	TTL_NOT_SET          = -1
-	TTL_KEY_DOESNT_EXIST = -2
+	TTL_NOT_SET             = -1
+	TTL_KEY_DOESNT_EXIST    = -2
+	disableExpiryWatcherEnv = "LOG_SERVICE_DISABLE_EXPIRY_WATCHER"
```
Comment (vistaarjuneja) [inline product/log-service/stream/redis/redis.go:50]: This should be added to the env variables and propagated through. We generally don't do os env lookups anywhere else in the code, it's all done at a common place

Comment (smjt-h) [reply product/log-service/stream/redis/redis.go]: added to config

File: product/log-service/config/config.go [hunk 2]
```diff
@@ -44,10 +44,11 @@ type Config struct {
...
 	Redis struct {
-		Endpoint   string `envconfig:"LOG_SERVICE_REDIS_ENDPOINT"`
-		Password   string `envconfig:"LOG_SERVICE_REDIS_PASSWORD" secret:"true"`
-		SSLEnabled bool   `envconfig:"LOG_SERVICE_REDIS_SSL_ENABLED"`
-		CertPath   string `envconfig:"LOG_SERVICE_REDIS_SSL_CA_CERT_PATH"`
+		Endpoint             string `envconfig:"LOG_SERVICE_REDIS_ENDPOINT"`
+		Password             string `envconfig:"LOG_SERVICE_REDIS_PASSWORD" secret:"true"`
+		SSLEnabled           bool   `envconfig:"LOG_SERVICE_REDIS_SSL_ENABLED"`
+		CertPath             string `envconfig:"LOG_SERVICE_REDIS_SSL_CA_CERT_PATH"`
+		DisableExpiryWatcher bool   `envconfig:"LOG_SERVICE_DISABLE_EXPIRY_WATCHER"`
```
Comment (vistaarjuneja) [inline product/log-service/config/config.go:51]: nit: `LOG_SERVICE_REDIS_DISABLE_EXPIRY_WATCHER`

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> Reviewed 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42151#-:-NKxGUvqCpnXgWJyN0Jw:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**